### DISCUSSION TITLE
questions about examples in sirius developer manual

### DISCUSSION THREAD

**Discussion by Willia-Yang** [eclipse-sirius/sirius-desktop#390] [Q&A]
the examples code provide in the manual dont include the dependencies need to be imported in the class, i dont konw what i should import exactly, can you give me the example's soucre code

**Comment by GlennPlou** [comment]
Hi,

Can you please specify the code examples you are using ?

Regards,
Glenn

**Reply by Willia-Yang** [reply]
hi, the code is a example about Providing custom StyleConfiguration in https://eclipse.dev/sirius/doc/developer/extensions-provide_custom_style.html

![图片](https://github.com/eclipse-sirius/sirius-desktop/assets/55096506/48b570c1-3cc8-4654-8efb-4cf7434cc67c)

**Reply by GlennPlou** [reply]
Note that you can easily resolve imports automatically using one of the Sirius targetplatforms in `releng/org.eclipse.sirius.targets/`.
For this example, you will use these imports:
```
import org.eclipse.sirius.diagram.description.DiagramElementMapping;
import org.eclipse.sirius.diagram.description.NodeMapping;
import org.eclipse.sirius.diagram.ui.tools.api.graphical.edit.styles.IStyleConfigurationProvider;
import org.eclipse.sirius.diagram.ui.tools.api.graphical.edit.styles.StyleConfiguration;
import org.eclipse.sirius.viewpoint.Style;
```
Constants and other instances are specific to the "DiagSeq" metamodel used in examples.

Regards,
Glenn

**Reply by Willia-Yang** [reply]
Sorry, I still don't how to import automatically by using the way you told me. I try to add it in dependencies or find it in my system explorer, but i can't find anything about `releng/org.eclipse.sirius.targets/`.
Regards.
